app/services/katello/content_plugin_manager.rb

```diff
@@ -0,0 +1,48 @@
+module Katello
+  class ContentPluginManager
+    @content_types = {}
+    class << self
+      attr_reader :content_types
+      private :new
+
+      def def_field(*names)
+        class_eval do
+          names.each do |name|
+            define_method(name) do |*args|
+              args.empty? ? instance_variable_get("@#{name}") : instance_variable_set("@#{name}", *args)
+            end
+          end
+        end
+      end
+      # Plugin constructor
+      def register(id, &block)
+        content_type = new(id)
+        content_type.instance_eval(&block)
+        content_types[id] = content_type
+      end
+
+      def repository_types
+        content_types.keys.map(&:to_s)
+      end
+
+      def creatable_by_user?(repository_type)
+        find(repository_type).allow_creation_by_user
+      end
+
+      def find(repository_type)
+        content_types[repository_type.to_sym]
+      end
+    end
+
+    def_field :description, :allow_creation_by_user
```

Comment:
@jlsherrill preferred this as this approach is consistent with fpremans plugin format -> https://github.com/theforeman/foreman/blob/fb9f45e819fbd15946084cd7cacd737dca5aa32e/app/services/foreman/plugin.rb#L42
 
I hate the above method too, but I was convinced that its better to stick to foreman's approach

Basically the above code creates a method
```
def  foo(..args)
  if args.empty? treat like a getter and retuen @foo
  else treat like a setter @foo-=args
end
```


If this is foremans plugin model code we should stick to it..